packages/qwik-nx/src/generators/preset/generator.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  addProjectConfiguration,
+  formatFiles,
+  generateFiles,
+  GeneratorCallback,
+  getWorkspaceLayout,
+  names,
+  offsetFromRoot,
+  Tree,
+} from '@nrwl/devkit';
+import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';
+import { QwikWorkspacePresetGeneratorSchema } from './schema';
+import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
+import * as path from 'path';
+import { getRelativePathToRootTsConfig } from '@nrwl/workspace/src/utilities/typescript';
+import { NormalizedSchema } from '../application/utils/normalize-options';
+import { configureVite } from '../application/generator';
+
+// In order to reuse the files in the application generator,
+// appProjectRoot = projectRoot
+interface PresetNormalizedSchema extends NormalizedSchema {
+  fileSrc: string;
+  projectDirectory: string;
+}
+
+function normalizeOptions(
+  tree: Tree,
+  options: QwikWorkspacePresetGeneratorSchema
+): PresetNormalizedSchema {
+  const name = names(options.name).fileName;
+  const projectDirectory = name;
+  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
+  const projectRoot = `${getWorkspaceLayout(tree).appsDir}/${projectDirectory}`;
+  const parsedTags = options.tags
+    ? options.tags.split(',').map((s) => s.trim())
+    : [];
+  const styleExtension = options.style !== 'none' ? options.style : null;
+  const fileSrc = '../application/files';
+
+  options.strict = options.strict ?? true;
+  options.unitTestRunner = options.unitTestRunner ?? 'vitest';
+
+  return {
+    ...options,
+    fileSrc,
+    projectDirectory,
+    projectName,
+    appProjectRoot: projectRoot,
+    offsetFromRoot: offsetFromRoot(projectRoot),
+    rootTsConfigPath: getRelativePathToRootTsConfig(tree, projectRoot),
+    parsedTags,
+    setupVitest: options.unitTestRunner === 'vitest',
+    styleExtension,
+  };
+}
+
+function addFiles(tree: Tree, options: PresetNormalizedSchema) {
```

Comment:
You shouldn't actually copy the logic from app generator, instead you can directly invoke it. For example here's `componentGenerator` being reused in library one https://github.com/qwikifiers/qwik-nx/blob/main/packages/qwik-nx/src/generators/library/generator.ts#L18